### PR TITLE
Use streaming download from upstream lfs server to avoid timeout

### PR DESCRIPTION
**Pull request by nick-mahony-wymac**

### CODE REVIEW

File: GitLfs.Client/FileCachingLfsClient.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@ namespace GitLfs.Client
     using System.IO;
     using System.Net.Http;
     using System.Net.Http.Headers;
-    using System.Threading.Tasks;
+		using System.Threading.Tasks;
```
Comment (glennawatson) [inline GitLfs.Client/FileCachingLfsClient.cs]: Mind keeping the indention consistent?

File: GitLfs.Core/File/LfsFileManager.cs [hunk 2]
```diff
@@ -100,6 +100,23 @@ public string SaveFile(string repositoryName, ObjectId objectId, FileLocation lo
             }
         }
 
+        /// <inheritdoc />
+        public Stream SaveFile(out string fileName, string repositoryName, ObjectId objectId, FileLocation location, Stream contents, string suffix = null)
+        {
+            Tuple<string, string> temporaryPath = this.GetDirectoriesAndFileNames(repositoryName, objectId, location, suffix);
```
Comment (glennawatson) [inline GitLfs.Core/File/LfsFileManager.cs]: Use C# 7 named tupes, you can do 

var (directory, fileName) = this.GetDirectoriesAndFileNames();

File: GitLfs.Core/File/CachingStream.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+using System.IO;
+
+namespace GitLfs.Core.File
+{
+	/// <summary>
+	/// Writes to an output file as an input stream is read
+	/// </summary>
+	internal class CachingStream : Stream
+	{
+		private readonly Stream _inStream;
```
Comment (glennawatson) [inline GitLfs.Core/File/CachingStream.cs]: Can you keep a consistent naming standard? I know wymac use _ but yeah.

File: GitLfs.Core/File/LfsFileManager.cs (hunk 2 above)
Comment (glennawatson) [inline GitLfs.Core/File/LfsFileManager.cs]: You can used named tuples, var (directory, filename) again.

**Comment by glennawatson** [comment]
Thanks for the bug fix :)

**Comment by glennawatson** [comment]
I should do a .editorconfig to make your guys easier. https://editorconfig.org/
Also setup some analyzers and stuff to avoid trivial stuff like that.

**Comment by glennawatson** [comment]
Put https://github.com/reactiveui/ReactiveUI/blob/master/.editorconfig in the main root directory where you are using VS code from. It will obey those settings from there on. Also Visual Studio 2017 (and 2015 with a plugin) obey that file. Might help with the VS code weirdness.

**Comment by nick-mahony-wymac** [comment]
I updated the indentation and naming but it was complaining about named tuples so I left it

**Comment by glennawatson** [comment]
Ah the named tuples would probably require you to upgrade the .NET Core to 2/2.1

**Comment by nick-mahony-wymac** [comment]
Thanks!

**Pull request merged**